reconcile/ocm_upgrade_scheduler.py

```diff
@@ -168,7 +208,19 @@ def version_conditions_met(version, history, ocm_name, workloads, upgrade_condit
     Returns:
         bool: are version upgrade conditions met
     """
-    conditions_met = True
+    # check if previous sectors run at least this version for that workload
+    # we will check dependencies recursively until there are versions for the given workload
+    # or no more dependencies to check
+    sector = upgrade_conditions.get("sector")
+    if sector:
+        for w in workloads:
+            for dep in workload_sector_dependencies(sector, w):
+                dep_versions = workload_sector_versions(dep, w)
```

Comment:
Do you  think a `lru` would make sense?

Reply:
I checked quickly and it does not seem to bring much value. It actually requires to rewrite some tests as it would wrongly cache some Sector args while dependencies do change in sequential tests.
I'd pass on this for now.